soweego/linker/linking_strategies.py

```diff
@@ -47,90 +50,134 @@ def baseline(source, target, strategy, output_dir):
-        _similar_names_wrapper(source_dataset, target_dataset, output_dir)
+        _perfect_name_wrapper(source_dataset, target_entity, output_dir)
+        _similar_names_wrapper(source_dataset, target_entity, output_dir)
+        _similar_links_wrapper(source_dataset, target_link_entity, output_dir)
 
 
 def _similar_names_wrapper(source_dataset, target_dataset, output_dir):
     LOGGER.info('Starting similar name match')
-    matches = similar_name_match(source_dataset, target_dataset)
+    matches = similar_name_match(
+        source_dataset, target_dataset, text_utils.tokenize)
     with open(path.join(output_dir, 'similar_name_matches.json'), 'w') as output_file:
         json.dump(matches, output_file, indent=2, ensure_ascii=False)
         LOGGER.info("Matches dumped to '%s'", output_file.name)
 
 
-def _similar_links_wrapper(source_dataset, target_dataset, output_dir):
+def _similar_links_wrapper(source_dataset, target_entity, output_dir):
     LOGGER.info('Starting similar link match')
-    matches = similar_link_match(source_dataset, target_dataset)
+    matches = similar_link_match(source_dataset, target_entity)
     with open(path.join(output_dir, 'similar_link_matches.json'), 'w') as output_file:
         json.dump(matches, output_file, indent=2, ensure_ascii=False)
         LOGGER.info("Matches dumped to '%s'", output_file.name)
     return matches
 
 
-def _perfect_string_wrapper(source_dataset, target_dataset, output_dir):
+def _perfect_name_wrapper(source_dataset, target_entity, output_dir):
     LOGGER.info('Starting perfect string match')
-    matches = perfect_string_match((source_dataset, target_dataset))
+    matches = perfect_name_match(source_dataset, target_entity)
     with open(path.join(output_dir, 'perfect_string_matches.json'), 'w') as output_file:
         json.dump(matches, output_file, indent=2, ensure_ascii=False)
         LOGGER.info("Matches dumped to '%s'", output_file.name)
 
 
-def perfect_string_match(datasets) -> dict:
+def perfect_name_match(source_dataset, target_entity: BaseEntity) -> dict:
     """Given an iterable of dictionaries ``{string: identifier}``,
     match perfect strings and return a dataset ``{id: id}``.
 
     This strategy applies to any object that can be
     treated as a string: names, links, etc.
     """
+    db_manager = DBManager()
+    session = db_manager.new_session()
     matched = {}
-    merged = defaultdict(list)
-    for dataset in datasets:
-        for string, identifier in dataset.items():
-            merged[string].append(identifier)
-    for string, identifiers in merged.items():
-        if len(identifiers) > 1:
-            LOGGER.debug("'%s': it's a match! %s -> %s",
-                         string, identifiers[0], identifiers[1])
-            matched[identifiers[0]] = identifiers[1]
+
+    for label, qid in source_dataset.items():
+        for res in session.query(target_entity).filter(target_entity.name == label).all():
```

Comment:
- TODO: move `connect_to_db` to `utils`;
- should use a safer construct here, as per `data_gathering.py`, i.e.:
```py
def connect_to_db():
    db_manager = DBManager()
    session = db_manager.new_session()
    return session

def DO_STUFF():
    session = connect_to_db()
    result = None
    try:
        result = RUN SOME QUERY HERE
        session.commit()
    except:
        session.rollback()
        raise
    finally:
        session.close()

    if not result:
        return None
    return PARSE_RESULT_FUNCTION(result)

```